lib/thrift/binary/framed/client.ex

```diff
@@ -116,33 +115,45 @@ defmodule Thrift.Binary.Framed.Client do
   def close(conn), do: Connection.call(conn, :close)
 
   @doc false
-  def connect(_, %{sock: nil, host: host, port: port, tcp_opts: opts, timeout: timeout, retries: retries, backoff_calculator: backoff_calculator} = s) do
+  def connect(info, %{sock: nil, host: host, port: port, tcp_opts: opts, timeout: timeout} = s) do
     opts = opts
     |> Keyword.merge(@immutable_tcp_opts)
     |> Keyword.put_new(:send_timeout, 1000)
 
     case :gen_tcp.connect(host, port, opts, timeout) do
       {:ok, sock} ->
-        {:ok, %{s | sock: sock, retries: 0}}
+        new_state = %{s | sock: sock}
+        retry_failed_message(info, new_state)
 
-      {:error, _} ->
-        new_retries = retries + 1
-        backoff = backoff_calculator.(new_retries)
+      {:error, _} = error ->
+        Logger.error("Failed to connect to #{host}:#{port} due to #{inspect error}")
+        {:stop, error, s}
+    end
+  end
 
-        msg = "Failed to connect to #{host} (after #{new_retries + 1} attempts), retrying in #{backoff}ms."
-        if retries <= 5 do
-          Logger.info(msg)
-        else
-          Logger.warn(msg)
-        end
+  defp retry_failed_message(:retry, %{retry: true, last_message: {call_args, caller}} = state) do
+    case handle_call(call_args, caller, state) do
+      {:reply, response, state} ->
+        Connection.reply(caller, response)
+        {:ok, %{state | last_message: nil}}
 
-        {:backoff, backoff, %{s | retries: new_retries}}
+      other ->
+        {:disconnect, {:error, other}, nil}
     end
   end
 
+  defp retry_failed_message(:retry, %{retry: false}) do
+    {:disconnect, {:error, :not_retrying}, nil}
```

Comment:
So previously where a client would get `{:error, :closed}`, they'll now get `{:error, :not_retrying}`? That might be confusing. It's unclear what's not being retried or why. Maybe `:connect_failed`? Or stick with `:closed`?

Also it would be good to see a test where this happens.